### PR TITLE
fix: filename too long in some cases

### LINKED ISSUE #75: Appmap generated file name too long
I tested on an existent Kotlin codebase and some tests started failing because the generated filename is too long. This is in part because the use of long descriptions in test function names is encouraged.

Running on MAC, I get the following:
```sh
$ getconf PATH_MAX /
1024
$ getconf NAME_MAX /
255
```

Perhaps we could truncate the file name when it approaches the limit.
An idea to then avoid filename clashes could be to include a md5 hash of the intend filename, when it would take more than 255 characters.

Stacktrace from v1.0.4:
```
com.appland.appmap.record.ActiveSessionException: failed to start recording session:
<full path here>appmap.json (File name too long)

        at com.appland.appmap.record.RecordingSessionFileStream.start(RecordingSessionFileStream.java:64)
        at com.appland.appmap.record.Recorder.setActiveSession(Recorder.java:44)
        at com.appland.appmap.record.Recorder.start(Recorder.java:87)
        at com.appland.appmap.process.hooks.ToggleRecord.startRecording(ToggleRecord.java:203)
        at com.appland.appmap.process.hooks.ToggleRecord.junitJupiter(ToggleRecord.java:246)
```

### PR DESCRIPTION
**Pull request by buzzcz**
This patch should similarly fix #75 as in https://github.com/getappmap/appmap-python/issues/91

### CODE REVIEW

File: agent/src/main/java/com/appland/appmap/record/Recorder.java [hunk 1]
```diff
@@ -1,18 +1,18 @@
 package com.appland.appmap.record;
 
-import java.io.IOException;
```
Comment (apotterri) [inline agent/src/main/java/com/appland/appmap/record/Recorder.java:3]: It doesn't look like most of these changes to the imports are necessary. Can you limit them to what's required for the new functionality, please?

Comment (buzzcz) [reply agent/src/main/java/com/appland/appmap/record/Recorder.java:3]: Oh, sorry about that. Default IDE settings for new project and I missed it. Fixed in the new version

File: agent/src/test/java/com/appland/appmap/record/RecorderTest.java [hunk 2]
```diff
@@ -239,4 +226,14 @@ public void testMultithreadCheckpoint() throws InterruptedException {
     es.shutdown();
   }
 
+  @Test
+  public void testSanitizeFilename() {
+    String longFilename = "foo".repeat(100);
+    assertEquals("foo.appmap.json", Recorder.sanitizeFilename("foo"));
+    assertEquals(
```
[the remark this answers is not in the document]
Comment (buzzcz) [reply agent/src/test/java/com/appland/appmap/record/RecorderTest.java]: That makes sense, I'll use it like this. Fixed in the next version

File: agent/src/test/java/com/appland/appmap/record/RecorderTest.java [hunk 3]
```diff
@@ -1,19 +1,17 @@
 package com.appland.appmap.record;
 
-import static com.github.stefanbirkner.systemlambda.SystemLambda.tapSystemErr;
```
Comment (apotterri) [inline agent/src/test/java/com/appland/appmap/record/RecorderTest.java:3]: Sorry to be a nuisance about this, but looks like some more import change snuck in.

**Review by apotterri — CHANGES_REQUESTED** [review]

**Review by apotterri — APPROVED** [review]
Looks good. Thanks again.

**Comment by buzzcz** [comment]
Perfect, thank you!

**Comment by kgilpin** [comment]
> Perfect, thank you!

Thanks so much for your contribution!

**Pull request merged**